modules/apps/portal/portal-monitoring/src/main/java/com/liferay/portal/monitoring/internal/servlet/filter/MonitoringFilter.java

```diff
@@ -227,6 +228,9 @@ private void _incrementProcessFilterCount() {
 	private static final Log _log = LogFactoryUtil.getLog(
 		MonitoringFilter.class);
 
+	private static final Snapshot<LayoutLocalService>
+		_layoutLocalServiceSnapshot = new Snapshot<>(
```

Comment:
@W2jx I think this can also be made into a hard ref.

Like `Portal`, this `LayoutLocalService` is also currently in core. However, being in core doesn't mean it will always start earlier than OSGi service in modules. During our startup, the Spring context may start concurrently with, or later than the module framework.

The actual reason that makes it safe to use a hard reference is that it's a filter. It will be executed only when someone hits the web page of portal.